yt/frontends/ramses/data_structures.py

```diff
@@ -857,9 +867,59 @@ def read_namelist(self):
 
             self.parameters["namelist"] = nml
 
+    @staticmethod
+    def _sanitize_filename(path):
+        # Do not forget to resolve so that it works with symlinks
+        filename = Path(path).resolve()
+
+        def check_standard_files(output_dir, iout):
+            # Check that the "amr_" and "info_" files exist
+            ok = (output_dir / f"amr_{iout}.out00001").is_file()
+            ok &= (output_dir / f"info_{iout}.txt").is_file()
+            return ok
+
+        def test_with_folder_name(output_dir):
+            iout_match = OUTPUT_DIR_RE.match(output_dir.name)
+            ok = output_dir.is_dir() and iout_match is not None
+            if ok:
+                iout = iout_match.group(2)
+                ok &= check_standard_files(output_dir, iout)
+                info_fname = output_dir / f"info_{iout}.txt"
+            else:
+                output_dir, info_fname = None, None
+
+            return ok, output_dir, info_fname
+
+        def test_with_standard_file(filename):
+            iout_match = OUTPUT_DIR_RE.match(filename.parent.name)
+            ok = (
+                filename.is_file()
+                and STANDARD_FILE_RE.match(filename.name) is not None
+                and iout_match is not None
+            )
+
+            if ok:
+                iout = iout_match.group(2)
+                ok &= check_standard_files(filename.parent, iout)
+                output_dir = filename.parent
+                info_fname = output_dir / f"info_{iout}.txt"
+            else:
+                output_dir, info_fname = None, None
+
+            return ok, output_dir, info_fname
+
+        for check_fun in (test_with_standard_file, test_with_folder_name):
+            ok, output_dir, info_fname = check_fun(filename)
+            if ok:
+                return output_dir, info_fname
+        else:
+            return None, None
+
     @classmethod
-    def _is_valid(self, *args, **kwargs):
-        if not os.path.basename(args[0]).startswith("info_"):
+    def _is_valid(self, filename, *args, **kwargs):
```

Comment:
Not related to this PR but there's a mistake here, might as well fix it. (also, this is the focus point of #2739)
```suggestion
    def _is_valid(cls, filename, *args, **kwargs):
```